docs/DeveloperGuide.md

```diff
@@ -299,89 +299,90 @@ The following activity diagram shows how a `Person` with `Status` and `Module` i
 Editing through `EditWindow` is largely similar to the above.
 <br /><br />
 
-### \[Proposed\] Undo/redo feature
+### Undo/Redo Feature
 
-#### Proposed Implementation
+#### Current Implementation
 
-The proposed undo/redo mechanism is facilitated by `VersionedAddressBook`. It extends `AddressBook` with an undo/redo history, stored internally as an `addressBookStateList` and `currentStatePointer`. Additionally, it implements the following operations:
+The undo/redo mechanism is facilitated by `StackUndoRedo`. The implemented undo/redo feature would be best described as two stacks of actions that the user has performed:
 
-* `VersionedAddressBook#commit()` — Saves the current address book state in its history.
-* `VersionedAddressBook#undo()` — Restores the previous address book state from its history.
-* `VersionedAddressBook#redo()` — Restores a previously undone address book state from its history.
+- The Undo stack serves to store a "history" of the actions they have performed.
+- The Redo stack is a collection of their actions that lead up to initial condition at which they started performing the undo.
 
-These operations are exposed in the `Model` interface as `Model#commitAddressBook()`, `Model#undoAddressBook()` and `Model#redoAddressBook()` respectively.
+They central concept is to store a stack of commands that essentially functions as a history-list of the commands. Essentially, we leverage on the stack's data structure of the which is a linear data structure that is based on the principle of Last In First Out (LIFO). Based on the implementation described above, the undo stack is populated by pushing a user's action in the application to the undo stack. 
```

Comment:
Possible typo? Should be "The central concept..." instead

Reply:
Good catch.